client/src/modules/Home/Components/Home.tsx

```diff
@@ -0,0 +1,119 @@
+import React, {useEffect, useState} from "react";
+import SearchBar from "../../../ui/SearchBar";
+import ProfileDropdown from "../../../ui/ProfileDropdown";
+import "../home.css";
+import {useAuthOptionalLogin} from "../../../auth/auth_utils";
+
+import DTITextLogo from "../../../assets/img/dti-text-logo.png";
+import DTIWhiteLogo from "../../../assets/img/dti-text-white-logo.png";
+
+/*
+  App Component. Uppermost View component in the component tree,
+  the first element of the HTML body tag grabbed by main.html.
+
+  Renders the application homepage with a navbar and searchbar, popular
+  classes and recent reviews components.
+*/
+export const Home = (imgSrc: any) => {
+  const [isLoggedIn, token, netId, signIn, signOut] = useAuthOptionalLogin();
+  const [DTILogo, setDTILogo] = useState(DTITextLogo);
+  const [season, setSeason] = useState("winter");
+  const [time, setTime] = useState("afternoon");
+
+  /** Logic for deciding the background image */
+  const setBackground = () => {
+    const sunset_start_times = [
+      17.0, 17.5, 18, 19.5, 20, 20.5, 20.5, 19.5, 18.5, 18, 16.5, 16.5,
+    ];
+    const sunset_end_times = [
+      18.5, 19, 20.5, 21, 22, 22, 22, 21.5, 20.5, 20, 19, 18,
+    ];
+    const date = new Date();
+    const month = date.getMonth();
+    const hours = date.getHours();
+    const minutes = date.getMinutes();
+
+    let time_of_day = hours;
+    if (minutes > 30) {
+      time_of_day += 0.51;
+    }
+    if (month === 11 || month === 0 || month === 1) {
+      setSeason("winter");
+    } else if (month >= 2 && month <= 4) {
+      setSeason("spring");
+    } else if (month >= 5 && month <= 7) {
+      setSeason("summer");
+    } else {
+      setSeason("fall");
+    }
+
+    if (time_of_day < 6 || time_of_day >= sunset_end_times[month]) {
+      setTime("night");
+    } else if (time_of_day >= sunset_start_times[month]) {
+      setTime("sunset");
+    }
+  };
+
+  useEffect(() => {
+    setBackground();
+    if (time === "night") {
+      setDTILogo(DTIWhiteLogo);
+    }
+  }, [time]);
+
+  /** Displays "sign in" or profile bear picture */
+  const NavButton = () => {
+    if (netId) {
```

Comment:
assuming that `netId` populates if user is signed in? Why not use `isLoggedIn` bool here for the if statement?

Reply:
Agreed, this was code that was developed after the isLoggedIn bool was created.

Reply:
Thanks for the catch. To be honest, I just copied and pasted the code from before but just refactored the logic behind it - did not really change this NavButton functionality-wise. I just updated it to use `isLoggedIn`. To be honest, I want to restructure the entire `useAuthOptionalLogin()` to return an object or a module with exports, not an array - which would also solve the unused variables warning. 

Will most likely do this in a next/future PR as long as it isn't too much.

Reply:
I agree, the `useAuthOptionalLogin()` is not implemented too well. This is a fix we can definitely make an add onto the list of tasks. I want to move away from Google login as it seems to be not as well maintained since Google seems to prefer just the use of Firebase authentication. We can take a look at other dependencies such as `Passport.js`. I'll do some research.